BC and GAIL (#3)
keisuke-nakata/baselines: general/chainerrl/README.md

# Getting started

### Dataset download

> - In order to run Behavoral Cloning and GAIL agents, you need to download expert dataset into an appropriate place (by default, `baselines/expert_dataset`).

Reviewer: ```suggestion
- In order to run the Behavoral Cloning and GAIL agents, you need to download the expert dataset into an appropriate location (by default, `baselines/expert_dataset`).
```

  4. `{'jump': 0, 'camera': [0, 10]}`  
  Note that `forward`, `sprint`, `attack` are always `1` and `back`, `left`, `right`, `sneak`, `place` are always `0`.
- Obtain*: `Discrete(36)`

# Experimental results of BC/GAIL

> The figures below show the *training* reward graphs for each of BC/GAIL with converting criteria of expert dataset.

Reviewer: ```suggestion
The figures below show the *training* reward graphs for both BC and GAIL along with the criteria used to process the expert dataset.
```

Videos of trained agents during their last evaluation round:
- [Behavioral Cloning trial 1 (reward 24.4)](static/release_bc_gail/BehavioralCloningNavigateDense1.mp4)
- [Behavioral Cloning trial 2 (reward 71.3)](static/release_bc_gail/BehavioralCloningNavigateDense2.mp4)
- [Behavioral Cloning trial 2 (reward 46.1)](static/release_bc_gail/BehavioralCloningNavigateDense3.mp4)
- [GAIL trial 1 (reward 9.8)](static/release_bc_gail/GAILNavigateDense1.mp4)
- (NA) GAIL trial 2
- [GAIL trial 3 (reward 54.2)](static/release_bc_gail/GAILNavigateDense3.mp4)

Notice: Since recording videos of BC during training phases is failed, we alternatively recorded after those phase are finished.

![Behavioral Cloning trial 2 first 100 frames](static/release_bc_gail/BehavioralCloningNavigateDense2.gif)  
Behavioral Cloning trial 2 first 100 frames

![GAIL trial 3 first 100 frames](static/release_bc_gail/GAILNavigateDense3.gif)  
GAIL trial 3 first 100 frames

## MineRLNavigate-v0

![release_bc_gail_MineRLNavigate-v0](static/release_bc_gail/MineRLNavigate-v0.png)

> BC and GAIL shows better performance than RL agents without expert dataset and [original paper](https://arxiv.org/abs/1904.10079)'s BC (4.23 +- 4.15).

Reviewer: ```suggestion
BC and GAIL show better performance than RL agents without expert dataset and [original paper](https://arxiv.org/abs/1904.10079)'s BC (4.23 +- 4.15).
```

Videos of trained agents during their last evaluation round:
- [Behavioral Cloning trial 1 (reward 0.0)](static/release_bc_gail/BehavioralCloningNavigate1.mp4)
- [Behavioral Cloning trial 2 (reward 100.0)](static/release_bc_gail/BehavioralCloningNavigate2.mp4)
- [Behavioral Cloning trial 3 (reward 100.0)](static/release_bc_gail/BehavioralCloningNavigate3.mp4)
- [GAIL trial 1 (reward 0.0)](static/release_bc_gail/GAILNavigate1.mp4)
- [GAIL trial 2 (reward 0.0)](static/release_bc_gail/GAILNavigate2.mp4)
- (NA) GAIL trial 3

![BehavioralCloning trial 2 first 100 frames](static/release_bc_gail/BehavioralCloningNavigate2.gif)  
Behavioral Cloning trial 2 first 100 frames

![GAIL trial 1 first 100 frames](static/release_bc_gail/GAILNavigate1.gif)  
GAIL trial 1 first 100 frames

## MineRLObtainDiamond-v0

> BC/GAIL does not solve `MineRLObtainDiamond-v0`, either.

Reviewer: ```suggestion
BC/GAIL aren't able to solve `MineRLObtainDiamond-v0`, either.
```
